stdlib/LinearAlgebra/src/generic.jl

```diff
@@ -449,11 +449,11 @@ diag(A::AbstractVector) = throw(ArgumentError("use diagm instead of diag to cons
 # Dot products and norms
 
 # special cases of norm; note that they don't need to handle isempty(x)
-generic_normMinusInf(x) = mapreduce(norm, min, x)
+generic_normMinusInf(x) = float(mapreduce(norm, min, x))
 
-generic_normInf(x) = mapreduce(norm, max, x)
+generic_normInf(x) = float(mapreduce(norm, max, x))
 
-generic_norm1(x) = mapreduce(norm, +, x)
+generic_norm1(x) = float(mapreduce(norm, +, x))
```

Comment:
Probably `mapreduce(float ∘ norm, +, x)`, otherwise you risk spurious overflow for small-integer types.